compiler/fir/resolve/src/org/jetbrains/kotlin/fir/types/ConeTypeIntersector.kt

```diff
@@ -145,8 +145,9 @@ object ConeTypeIntersector {
         protected fun ConeKotlinType.resultNullability(context: ConeTypeContext): ResultNullability =
             when {
                 isMarkedNullable -> ACCEPT_NULL
+                this is ConeFlexibleType -> UNKNOWN
```

Comment:
I think this should be `this is ConeFlexibleType -> upperBound.resultNullability(context)` instead. Otherwise, a `@NotNull List<...>` from Java would map to `ft(MutableList<...>..List<...>)`, which then would yield unknown nullability, despite it should be not null.

Reply:
On one hand, upper bound's nullability is more accurate, for sure, which requires one more recursive call here. On the other hand, `UNKNOWN` seems a safe approximation too, since as long as the upper bound is involved in intersection computation (and it indeed is), we'll get the same nullability at the end.

So, I updated the 1st commit to mentioned that aspect as code comments. 2nd commit updated this place as per the suggestion, along with the attempt to trigger such case (not successful, though). I'll let @demiurg906 take a look. Feel free to drop 2nd commit; squash two; or leave them as-is.